File: backend/core/models/llm/llmStatus.py
# pylint: disable=invalid-name
"""Modèle métier pour le statut du service LLM."""
from dataclasses import dataclass
from typing import Optional

from backend.core.models.llm.installedModel import InstalledModel  # noqa: E402


@dataclass
class LLMStatus:
    """Statut détaillé du service LLM."""
    available: bool
    message: str
    resources_path: str
    resolved_model_path: Optional[str]
    resolved_model_exists: Optional[bool]
    selected_model_id: Optional[str]
    installed_models: list[InstalledModel]

File: desktop/src/renderer/models/llm/LlmStatus.ts
import type { InstalledModel } from "./InstalledModel";

export interface LlmStatus {
  available: boolean;
  message: string;
  resolved_model_path: string | null;
  resolved_model_exists: boolean | null;
  resources_path: string | null;
  selected_model_id: string | null;
  installed_models: InstalledModel[];
}
